src/test/kotlin/uk/gov/justice/digital/hmpps/subjectaccessrequestworker/services/PdfServiceTest.kt

```diff
@@ -0,0 +1,178 @@
+package uk.gov.justice.digital.hmpps.subjectaccessrequestworker.services
+
+import com.google.gson.Gson
+import com.itextpdf.kernel.pdf.PdfDocument
+import com.itextpdf.kernel.pdf.PdfReader
+import com.itextpdf.kernel.pdf.canvas.parser.PdfTextExtractor
+import com.itextpdf.layout.Document
+import com.microsoft.applicationinsights.TelemetryClient
+import org.apache.commons.lang3.StringUtils
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.extension.ExtendWith
+import org.junit.jupiter.api.io.CleanupMode
+import org.junit.jupiter.api.io.TempDir
+import org.mockito.Mock
+import org.mockito.junit.jupiter.MockitoExtension
+import uk.gov.justice.digital.hmpps.subjectaccessrequestworker.models.DpsService
+import uk.gov.justice.digital.hmpps.subjectaccessrequestworker.models.SubjectAccessRequest
+import uk.gov.justice.digital.hmpps.subjectaccessrequestworker.services.pdf.PdfParameters
+import uk.gov.justice.digital.hmpps.subjectaccessrequestworker.services.pdf.PdfService
+import java.io.ByteArrayOutputStream
+import java.io.File
+import java.io.InputStreamReader
+import java.nio.file.Files
+import java.nio.file.Path
+import java.time.LocalDate
+import java.util.UUID
+
+@ExtendWith(MockitoExtension::class)
+class PdfServiceTest {
+
+  private lateinit var legacyPdfService: GeneratePdfService
+  private lateinit var newPdfService: PdfService
+
+  @TempDir(cleanup = CleanupMode.ALWAYS)
+  private lateinit var tempDir: Path
+
+  @Mock
+  private lateinit var telemetryClient: TelemetryClient
```

Comment:
you can use mockito kotlin to mock the telemetry client like
  private val telemetryClient: TelemetryClient = mock()